### ISSUE TITLE
Reverse proxy fails with 500 on a that allows client certificates

### ISSUE THREAD

**Issue by SierraNL** [Azure/service-fabric-issues#1432]
After changing all the services in our SF application to HTTPS, we also wanted to add authentication using client certificates. This works for communication between the services, but once you switch a service to allow or require client certificates, the reverse proxy return's a http status code 500 and doesn't seem to log anything.

## Expected Behavior
If the service allows client certificates, the reverse proxy should be able to ignore it and just proxy the required service. But if you switch to required, I would expect the reverse proxy, to return the 403 forbidden.

## Current Behavior
The reverse proxy returns a http status code 500 when trying to proxy a request to a service that allows or requires client certificates

## Steps to Reproduce
1. Switch the reverse proxy to HTTPS (https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reverseproxy-configure-secure-communication#secure-connection-establishment-between-the-reverse-proxy-and-services)
2. Add an application with a service that has an https endpoint and has a working page on it
3. Change the client certificate option on the service to allow
For instance in a dotnet core service's kestrel configuration:
```csharp
options.ListenAnyIP(port, listenOptions =>
                {
                    var adapterOptions = new HttpsConnectionAdapterOptions
                    {
                        ClientCertificateMode = ClientCertificateMode.AllowCertificate
```
4. Deploy the service to the cluster
5. Call it using https://localhost:19081/application/service/endpoint

## Context (Environment)
We're building an application and wanted to combine a switch to secure services with HTTPS and authentication using client certificates. And knowing the limitation of the reverse proxy, that it can only pass a clientcertificate in a header, and it isn't possible to configure a specific client certificate for the reverse proxy, we wanted our services to accept both an actual client certificate and a certificate in a header coming from the reverse proxy.

But this issue makes this scenario impossible.

#### Service Fabric Runtime and SDK Version :
Runtime: 6.4.622.9590
SDK: 3.3.622.9590

#### Operating System :
Windows 10 

#### Cluster Size :
Unsecured dev cluster, 1 node

## Possible Workaround
1. Don't use clientcertificate authentication within the cluster, or don't use the reverse proxy to access these services.

2. Make it possible to configure a client certificate for the reverse proxy itself.

**Comment by FesterCluck** [comment]
You need a tunnel, not a proxy. See if you can emulate whichever of these makes the most sense: https://en.wikipedia.org/wiki/Tunneling_protocol.

Reverse proxies are internal facing impersonators. They necessarily alter traffic, not encapsulate it. You're getting a 500 because the proxy doesn't understand the data it is trying to inspect and change (because it's encrypted using keys it has no access to).

tl;dr: What you are designing introduces MITM when implemented right. The standard reverse proxy probably isn't the place for this. Consider tunnels.

**Comment by SierraNL** [comment]
The propose of the Service Fabric proxy is to locate on which node a service resides. So if you're connecting to a service from outside the service fabric cluster and the service is not configured to run on every node, you need the reverse proxy to connect to this service on the correct node. That is why we're using it in the first place.

I'm reporting this as an issue because the reverse proxy should be able to connect to a service which is hosted on HTTPS and accepts client certificates. Since it's optional.

I really don't know what you're trying to say about keys it has no access to. It connects fine to a service on HTTPS, just not when it requests a client certificate (which is again, optional).

The connection from the reverse proxy to the service is an HTTPS connection already, so I really don't have a clue on how I introduce MITM with this design.

**Comment by FesterCluck** [comment]
My apologies, I was confusing auth and encryption. Have you attempted the solution provided at https://stackoverflow.com/questions/46416978/service-fabric-reverse-proxy-with-certificates-on-endpoint
